tests/test_config_flag.py

```diff
@@ -29,6 +30,44 @@ def test_deepmerge_dicts() -> None:
     }
 
 
+def test_specific_overrides_general() -> None:
```

Comment:
This should technically be part of a `test_registry.py`, which will contain other tests in the future. However, there should be a test for `_flatten_config` here.